src/commands/importFunction/importFunctionApp.ts

```diff
@@ -0,0 +1,405 @@
+    options.method = "POST";
+    options.url = `${baseUrl}/host/default/listkeys?api-version=${Constants.functionAppApiVersion}`;
+    options.headers = {
+        ['User-Agent']: appendExtensionUserAgent()
+    };
+    await signRequest(options, credentials);
+    // tslint:disable-next-line: await-promise
+    const funcKeys = await request(options).promise();
+    // tslint:disable-next-line: no-unsafe-any
+    return JSON.parse(funcKeys);
+}
+
+// create new operation for each function and method
+function getNewOperation(apiId: string, method: string, name: string, route: string, displayName: string): OperationContract {
+    const operationId = apiUtils.displayNameToIdentifier(name);
+    const cleanUrl = Utils.parseUrlTemplate(route);
+    return { id: `${apiId}/operations/${operationId}`, name: operationId, displayName: displayName, method: method, urlTemplate: cleanUrl.urlTemplate, description: "", templateParameters: cleanUrl.parameters };
+}
+
+function getFunctionAppBase(invokeUrlTemplate: string, route: string): string {
+    const uri = invokeUrlTemplate.substr(0, invokeUrlTemplate.length - route.length);
+    return uri.endsWith("/") ? uri.substr(0, uri.length - 1) : uri;
+}
+
+async function createApiFromFunctionApp(apiId: string, funcApp: Site, apiName: string): Promise<ApiContract> {
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Create Api for the function app ${funcApp.name}...`));
+    if (funcApp.name) {
+        return {
+            description: `Import from "${funcApp.name}" Function App`,
+            id: apiId,
+            name: apiName,
+            displayName: apiName,
+            path: "",
+            protocols: ["http"]
+        };
+    } else {
+        throw new Error("Unexpected null value for this function app : name");
+    }
+}
+
+// Create client for webiste mgmt client
+function getClient(node: ApisTreeItem | ApiTreeItem): WebSiteManagementClient {
+    const clientInfo: IAzureClientInfo = {
+        credentials: node.root.credentials,
+        subscriptionId: node.root.subscriptionId,
+        environment: node.root.environment
+    };
+    return createAzureClient(clientInfo, WebSiteManagementClient);
+}
+
+// Get all functions from a function app
+async function listAllFunctions(baseUrl: string, credentials: ServiceClientCredentials): Promise<IFunctionContract[]> {
+    const queryUrl = `${baseUrl}/functions?api-version=${Constants.functionAppApiVersion}`;
+    const funcAppObj = await requestUtilWithCredentials(queryUrl, credentials);
+    // tslint:disable-next-line: no-unsafe-any
+    return JSON.parse(funcAppObj).value;
+}
+
+// Get new api id
+async function getNewApiName(func: string): Promise<string> {
+    const inputName = await askApiName(func);
+    if (inputName === undefined && inputName === "") {
+        return func;
+    } else {
+        return inputName;
+    }
+}
+
+// User input api Name
+async function askApiName(funcName: string): Promise<string> {
```

Comment:
Remember seeing this in apiUtils as well

Reply:
yep, I took the same name, but the functionality is different, let me put it into apiUtil and rename it